File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/city.ts
export interface ICity{
    id? : number;
    stateId? : string;
    name? : string;
    county? : string;
}

export class City implements ICity {
    id? : number;
    stateId? : string;
    name? : string;
    county? : string;

    public constructor(init?:Partial<ICity>) {
        Object.assign(this, init);
    }
}

File: Rivington.IG.Web/Rivington.IG.Domain.Models/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Rivington.IG.Domain.Models
{
    public class City
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Name { get; set; }
        public string County { get; set; }

        public string StateId { get; set; }
        [ForeignKey(nameof(StateId))]
        public State State { get; set; }
    }
}